wordpress-mobile/WordPress-Android
# Login rework: email input screen

Introduces the new login flow's email input screen.

This PR includes work on the visual style of the screen but the final touches will be performed in a future PR.

At the moment, the screen doesn't implement any functionality that moves away from the email screen, other than getting back to the prologue screen.

There's been a rather important change to the implementation of the Navigation State Machine with the introduction of a `Stack` to hold the history of the screens instead of just the latest screen. This also facilitates going _back_ in the history.

The navigation state is moved into a headless, retained Fragment (`LoginNavFragment`). The parent Activity is now cleaner, but still plays the role of the "proxy" between Fragment communication.

To test:
* Open the app while being logged out
* Tap on the "Log in" button on the prologue screen
* The email input screen should show up
* Try putting in some text/email in the EditText and observe the various error cases
* "Help", "Next" and "Log in with site address instead" do not work for now, only show a Toast when a change to another screen would occur
